src/coreclr/jit/importer.cpp

```diff
@@ -4515,13 +4515,54 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             case NI_System_Math_Log:
             case NI_System_Math_Log2:
             case NI_System_Math_Log10:
-#ifdef TARGET_ARM64
+#if defined TARGET_ARM64
             // ARM64 has fmax/fmin which are IEEE754:2019 minimum/maximum compatible
             // TODO-XARCH-CQ: Enable this for XARCH when one of the arguments is a constant
             // so we can then emit maxss/minss and avoid NaN/-0.0 handling
             case NI_System_Math_Max:
             case NI_System_Math_Min:
 #endif
+
+#if defined FEATURE_HW_INTRINSICS && defined TARGET_XARCH
+            case NI_System_Math_Max:
+            case NI_System_Math_Min:
+            {
+                assert(varTypeIsFloating(callType));
+
+                if (sig->numArgs == 2 && (impStackTop().val->IsCnsFltOrDbl() || impStackTop(1).val->IsCnsFltOrDbl()))
+                {
+                    GenTree* op2 = impPopStack().val;
+                    GenTree* op1 = impPopStack().val;
+
+                    unsigned int insSimdSize = (callType == TYP_DOUBLE) ? 32 : 16;
+
+                    // IEEE 754 spec requires:
+                    // 1. If both operands are 0 of either sign
+                    // IsFloatPositiveZero is used for both operands, because the sign doesn't matter
+                    // 2. If either of operands is NaN (handles if both operands are NaN too)
+                    // 3. If either of operands is (-0.0) and an instruction is Math.Min
+                    // 3. If either of operands is (+0.0) and an instruction is Math.Max
+                    // Then op2 is returned;
```

Comment:
Because of this, we can only specially fold things if at least one operand is constant, because then we can account for how the underlying instruction behaves.

For `Max`:
* When one operand is constant and `-0` we can do `max -0, other`
* When one operand is constant and `+0` we cannot fold
  * If other was `NaN` we'd need `max +0, other`
  * If other was `+0` we'd need `max other, +0`
* When one operand is constant and `NaN` we can just return `NaN`
* When one operand is constant and any other value we can do `max cns, other` or `max other, cns`

For `Min`:
* When one operand is constant and `+0` we can do `min +0, other`
* When one operand is constant and `-0` we cannot fold
  * If other was `NaN` we'd need `min -0, other`
  * If other was `+0` we'd need `min other, -0`
* When one operand is constant and `NaN` we can just return `NaN`
* When one operand is constant and any other value we can do `min cns, other` or `min other, cns`